### PR TITLE
Update GH action to build and test on multiple OSes

### PR DESCRIPTION
**Pull request by vcheung-stripe**
We were running this on ubuntu-latest only. Updated to run on ubuntu-latest, macos-latest, and windows-latest. Also updated to run on pushes and PRs to all branches instead of only master.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
+          node-version: ${{ env.NODE_VERSION }}
 
-    - name: Use Node.js ${{ env.NODE_VERSION }}
-      uses: actions/setup-node@v1
-      with:
-        node-version: ${{ env.NODE_VERSION }}
+      - name: npm install, and test (ubuntu)
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: |
+          npm install
```
Comment (suz-stripe) [inline .github/workflows/build.yml:23]: extremely minor nit but we could dry this up by making `npm install` its own step before branching out into the different OS steps for the test command

**Review by suz-stripe — APPROVED** [review]
this LGTM as is! 👍 
I'll leave it up to you as to whether you wanted to roll my suggestion into this PR as it's super minor

File: .github/workflows/build.yml (hunk 1 above)
Comment (vcheung-stripe) [reply .github/workflows/build.yml:23]: Good suggestion, I'll do that.

**Review by gracegoo-stripe — APPROVED** [review]
Thank you so much for adding this!

**Pull request merged**